etcdserver/api/membership/cluster.go

```diff
@@ -59,6 +59,16 @@ type RaftCluster struct {
 	// removed contains the ids of removed members in the cluster.
 	// removed id cannot be reused.
 	removed map[types.ID]bool
+
+	downgradeInfo *DowngradeInfo
+}
+
+type DowngradeInfo struct {
+	// TargetVersion is the target downgrade version, if the cluster is not under downgrading,
+	// the targetVersion will be nil
+	TargetVersion *semver.Version
```

Comment:
Should we tag json fields and do we need import `semver` package? Can we just define its own field, and create `semver.Version` object once read from storage later?

Reply:
Change the type of `TargetVersion` to `string` and verify the version format when reading from backend.